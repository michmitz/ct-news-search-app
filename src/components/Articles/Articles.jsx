import React from 'react';
import PropTypes from 'prop-types';
import Article from '../Article/Article';

const Articles = ({ articles }) => {
  const articleslist = articles.map(article => (
    <li key={article.title}>
        <Article {...article} />
    </li>
  ));

  return (
    <ul data-testid="articles-list">
      {articleslist}
    </ul>
  );
};

Articles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  })).isRequired
};

export default Articles;