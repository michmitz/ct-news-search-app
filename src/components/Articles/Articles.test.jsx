import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Articles from './Articles';

describe('Articles component', () => {
  afterEach(() => cleanup());
  it('renders Articles List', () => {
    const { asFragment } = render(<Articles
      articles={[
        {
          title: 'test-title',
          author: 'test-name',
          description: 'test-description'
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
