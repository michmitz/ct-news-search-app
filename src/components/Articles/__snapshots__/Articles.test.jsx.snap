// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component renders Articles List 1`] = `
<DocumentFragment>
  <ul
    data-testid="articles-list"
  >
    <li>
      <h1>
        test-title
      </h1>
      <h3>
        test-name
      </h3>
      <p>
        test-description
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
